src/api/entities/Agent.ts

```diff
@@ -34,4 +41,29 @@ export class Agent extends Identity {
 
     this.ticker = ticker;
   }
+
+  /**
+   * Retrieve the agent group associated with this Agent
+   */
+  public async getPermissionGroup(): Promise<CustomPermissionGroup | KnownPermissionGroup> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { externalAgents },
+        },
+      },
+      context,
+      ticker,
+      did,
+    } = this;
+
+    const rawTicker = stringToTicker(ticker, context);
+    const rawIdentityId = stringToIdentityId(did, context);
+
+    const rawGroupPermissions = await externalAgents.groupOfAgent(rawTicker, rawIdentityId);
+
+    const agentGroup = rawGroupPermissions.unwrap();
```

Comment:
This could be `None`. Throw an error in that case. Something like "This agent is no longer part of a permission group"

Reply:
Let's change the text to "This Identity is no longer an Agent for this Security Token"